compiler/src/dmd/globals.h

```diff
@@ -196,6 +195,9 @@ struct Param
     Help help;
     Verbose v;
 
+    unsigned short edition;      // edition year
+    void* editionFiles;          // Edition corresponding to a filespec
```

Comment:
May have to think about some helper functions for C++ to interact with this D assoc array.

Else I guess gdc/ldc are left out of the CLI support.